build: Place CILIUM_ENVOY_SHA dependency in the Dockerfile

jrajahalme opened the pull request:
Automatically triggered builds (such as cilium/cilium:latest) can't
run "make" to start the docker build, but must build from the
Dockerfile without any prepatory steps. Enable this by moving the
CILIUM_ENVOY_SHA dependency to the Dockerfile instead of a separate
file. Makefile.defs still defines CILIUM_ENVOY_SHA as a variable, but
gets it from the Dockerfile.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/6158)
<!-- Reviewable:end -->

ianvernon commented on pkg/envoy/envoy.go:
```diff
@@ -39,10 +39,9 @@ import (
-	// This value must match the one in file CILIUM_ENVOY_SHA.
-	RequiredEnvoyVersionSHA = "fd3866831d8f85e27120f34d98133343fc9c4e0e"
+	// RequiredEnvoyVersionSHA is set during build
+	RequiredEnvoyVersionSHA string
```
Can you keep the log message about how `cilium-agent` will fail to start if there is a version mismatch?

jrajahalme replied:
Done

jrajahalme committed:
.gitignore: Ignore temporary LLVM files

Now that we build docker images without cleaning, we must exclude also
temporary files generated by a parallel build. This helps avoid errors
like this:

error checking context: 'file ('/home/vagrant/go/src/github.com/cilium/cilium/bpf/bpf_netdev.ll-8d2f3a79')
not found or excluded by .dockerignore'.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

jrajahalme committed:
build: Place CILIUM_ENVOY_SHA dependency in the Dockerfile

Automatically triggered builds (such as cilium/cilium:latest) can't
run "make" to start the docker build, but must build from the
Dockerfile without any prepatory steps. Enable this by moving the
CILIUM_ENVOY_SHA dependency to the Dockerfile instead of a separate
file. Makefile.defs still defines CILIUM_ENVOY_SHA as a variable, but
gets it from the Dockerfile.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>
